src/NuGet.Core/NuGet.Configuration/Settings/ISettings.cs

```diff
@@ -11,52 +11,41 @@ namespace NuGet.Configuration
     /// </summary>
     public interface ISettings
     {
-        /// <summary>
-        /// Folder under which the config file is present
-        /// </summary>
-        string Root { get; }
```

Comment:
similar to APIs there might be usage of these members as well, so why not mark them obsolete as well?

Reply:
`ISettings` used to be not only the computed settings but also a single nuget.config, therefore it has a pointer to the `root` and `filename`. In the new design, `ISettingsFile` is the pointer to a single nuget.config, but `ISettings` only manages the computed values. If we only mark them as deprecated, what values should the implementation return?

Reply:
`ISettings` used to point to **one specific** nuget.config, and had pointers to the next nuget.config, thus depending on which `ISettings` you called you could have gotten a different setting value (most of out higher level APIs exposed only one of this configs). The new `ISettings` points to **n** nuget.config (i.e. the computed settings), therefore there is no reference to "the same value" as the previous design. What `root` and `filename` of which config do you think we should be showing if we left this option?

If we are really concerned about deleting this, we could mark them obsolete with `error = true` and have the implementation just return an empty string, but that would have the same "breaking" effect for anyone that is actually using this (which I don't expect to be anyone, since `SettingsUtility` has utility methods to extract the `root` or `filename` of all `SettingsFile` for a given `ISetting`)

Reply:
Discussed offline, if @rrelyea is fine with breaking existing usage of these interface in 15.9 then I'm fine too.